webpack/webpack5-module-minifier-plugin/src/AsyncImportCompressionPlugin.ts

```diff
@@ -0,0 +1,314 @@
+                  localAsyncImports.set(key, {
+                    meta,
+                    module: targetModule
+                  });
+                }
+              }
+            });
+          }
+        }
+
+        const rankedImports: [string, IAsyncImportMetadata][] = [...asyncImportGroups]
+          .filter((x) => x[1].count > 1)
+          .sort((x, y) => {
+            let diff: number = y[1].count - x[1].count;
+            if (!diff) {
+              diff = y[1].chunkCount - x[1].chunkCount;
+            }
+
+            if (!diff) {
+              diff = x[0] > y[0] ? 1 : x[0] < y[0] ? -1 : 0;
+            }
+
+            return diff;
+          });
+
+        for (let i: number = 0, len: number = rankedImports.length; i < len; i++) {
+          rankedImports[i][1].index = i;
+        }
+
+        rankedImportGroups = rankedImports.map((x) => x[1]);
+
+        const { dependencyTemplates } = compilation;
+
+        const defaultImplementation: IImportDependencyTemplate | undefined =
+          dependencyTemplates.get(ImportDependency);
+
+        if (!defaultImplementation) {
+          compilation.errors.push(new WebpackError(`Could not find ImportDependencyTemplate`));
+        }
+
+        const customTemplate: IImportDependencyTemplate = {
+          apply(
+            dep: WebpackImportDependency,
+            source: sources.ReplaceSource,
+            dependencyTemplateContext: DependencyTemplateContext
+          ): void {
+            const targetModule: Module = moduleGraph.getModule(dep);
+
+            if (targetModule) {
+              const moduleId: string | number = chunkGraph.getModuleId(targetModule);
+              const stringKey: string = `${moduleId}`.replace(/[^A-Za-z0-9_$]/g, '_');
+              const key: string = `${ASYNC_IMPORT_PREFIX}${stringKey}`;
+              const content: string = `__webpack_require__.ee(${key})`;
+              source.replace(dep.range[0], dep.range[1] - 1, content);
+            } else {
+              defaultImplementation?.apply(dep, source, dependencyTemplateContext);
+            }
+          }
+        };
+
+        dependencyTemplates.set(ImportDependency, customTemplate);
+      });
+
+      compilation.mainTemplate.hooks.requireExtensions.tap(PLUGIN_NAME, (source, chunk) => {
+        if (!needChunkOnDemandLoadingCode(chunk)) {
+          return source;
+        }
+
+        const { requireFn } = compilation.mainTemplate;
+        return Template.asString([
+          `var asyncImportChunkGroups = [`,
+          rankedImportGroups
+            ? rankedImportGroups.map((x) => Template.indent(JSON.stringify(x.chunkIds))).join(',\n')
+            : '',
+          `];`,
+          `${requireFn}.ee = function (groupOrId, moduleId, importType) {`,
+          Template.indent([
+            `return Promise.all((Array.isArray(groupOrId) ? groupOrId : asyncImportChunkGroups[groupOrId]).map(function (x) { return ${requireFn}.e(x); }))`,
```

Comment:
There's a `simpleFunction` helper to emit functions or lambdas depending on target.